[add] 컬럼별 조회기능 1차 구현

happy-dev-travel opened the pull request:
- 컬럼별로 조회가 가능합니다. 간단한 사용방법은 테스트 코드 확인해 주시고, 현재 employeeNum으로 하는 검색은 findEmployee에 구현되어 있습니다. 
- Employee 클래스의 필드가 축소되었으니 확인 부탁드립니다

happy-dev-travel committed:
[add] 컬럼별 조회기능 1차 구현
- 컬럼별로 조회가 가능합니다. 간단한 사용방법은 테스트 코드 확인해 주시고, 현재 employeeNum으로 하는 검색은 findEmployee에 구현되어 있습니다.

jw451 commented on src/main/java/command/sch/SearchCommand.java:
```diff
@@ -113,17 +113,17 @@ private boolean isSearchCmd(UserRequest request) {
-        for (Map.Entry<String, Employee> entry : db.nameMap.get(key).entrySet()) {
-            resultList.add(entry.getValue());
-        }
+//        for (Map.Entry<String, Employee> entry : db.nameMap.get(key).entrySet()) {
```
고생하셨습니다!.이 주석은 삭제해도 되지않을까요? :)

happy-dev-travel replied:
아, 네네 아마 성우님이 SearchCommand 개발하시면서 이쪽 부분이 다 정리될 것 같습니다 :)

junheui27 commented on src/main/java/database/EmployeeDB.java:
```diff
@@ -7,23 +7,15 @@
-    private HashMap<CERTI, HashMap<String,Employee>> tableByCERTI = new HashMap<>();
-
-
+    final private HashMap<String, Employee> idHash = new HashMap<>();
+    final private HashMap<String, HashMap<String,Employee>> nameHash = new HashMap<>();
+    final private HashMap<String, HashMap<String,Employee>> phoneNumberHash = new HashMap<>();
+    final private HashMap<String, HashMap<String,Employee>> careerLevelHash = new HashMap<>();
+    final private HashMap<String, HashMap<String,Employee>> birthHash = new HashMap<>();
+    final private HashMap<String, HashMap<String,Employee>> certiHash = new HashMap<>();
```
여기서 final 을 사용하는 이유는 어떤 이유인가요??

happy-dev-travel replied:
일단.. intelliJ의 추천이 있었고요ㅎㅎ final을 사용하면 다른 곳에서 다시 생성하지 못하도록 막아줍니다

junheui27 commented on src/main/java/model/Employee.java:
```diff
@@ -11,50 +11,27 @@
 
     public Employee Copy(){
-        Employee copied = new Employee().builder()
+        return new Employee().builder()
```
builder 생성시 매개변수도 줄고 코드도 훨씬 간결해져서 보기 편한 것 같습니다.

jw451 reviewed (approved):
필드가 많이 간소화되었군요. 고생하셨습니다.
이를 바탕으로 printer 수정 및 테스트해보겠습니다.

smilesw6448 commented on src/main/java/database/EmployeeDB.java:
```diff
@@ -69,14 +109,41 @@ public Employee deleteEmployee(Employee deleteEmployee){
         return new Employee();
     }
 
     public List<Employee> findEmployeeByColumn(String columnName, String value){
```
employee 타입의 List가 리턴되는건 사용자 입장에서 너무 좋은거 같습니다.

그런데 해당 함수에 '사번' 으로 찾는 로직이 빠진것 같습니다.

smilesw6448 replied:
추가 설명을 드리자면, Search에서 employ로 검색호출시에는 타 함수로 호출을 해야하나요? 
findEmployee

happy-dev-travel replied:
사번으로 찾는 함수는 findEmployee 이고요, findEmployeeByColumn에서 지원하지 못하는 이유는 데이터 형식이 달라서 그렇긴 한데.. 음 합쳐볼까요

smilesw6448 replied:
User입장에서 데이터형식이 다른걸 (뭐가 어떤 HASH 인지) 몰라야 하기 때문에,
동일 함수를 호출하는 것이 맞지 않나 싶어서요

happy-dev-travel replied:
네네 알던 반영하였습니다 그런데 id로 검색하는 경우는 항상 반환되는 employee의 갯수가 1개 or 0개여서 별도의 함수를 쓰는게 더 좋지 않을까 생각합니다 :)

smilesw6448 replied:
제가 너무 나간거 같은 생각이 듭니다만 ㅠ.ㅠ
혹시라도 추후에 특정 사번(16사번, 17사번 등) 만 검색하라는 요청이 왔을 경우에
같인 함수를 써야할거 같아서요...!

smilesw6448 reviewed (changes requested):
findEmployeeByColumn 함수에 사번으로 검색하는 경우는 빠져있는 것 같습니다.

happy-dev-travel committed:
[add] 컬럼별 조회기능 1차 구현
- 컬럼별로 조회가 가능합니다. 간단한 사용방법은 테스트 코드 확인해 주시고, 현재 employeeNum으로 하는 검색은 findEmployee에 구현되어 있습니다.

- 성우님 의견 반영하여 findEmployeeByColumn에서도 employeeNum으로 검색할 수 있도록 변경

smilesw6448 commented on src/main/java/database/EmployeeDB.java:
```diff
@@ -32,56 +23,141 @@ public boolean addEmployee(Employee newEmployee){
 
     public List<Employee> findEmployeeByColumn(String columnName, String value){
-        return new ArrayList<>();
+        if(columnName.equals("employeeNum")){
```
확인했습니다~!
